backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -798,22 +844,33 @@ public Project getProjectByID(String id) throws InvalidProtocolBufferException {
   public Project setProjectShortName(String projectId, String projectShortName, UserInfo userInfo)
       throws InvalidProtocolBufferException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
-      Query query = session.createQuery(GET_PROJECT_BY_SHORT_NAME_AND_OWNER_HQL);
-      query.setParameter("projectShortName", projectShortName);
-      query.setParameter("vertaId", authService.getVertaIdFromUserInfo(userInfo));
-      ProjectEntity projectEntity = (ProjectEntity) query.uniqueResult();
-      if (projectEntity != null) {
-        Status status =
-            Status.newBuilder()
-                .setCode(Code.ALREADY_EXISTS_VALUE)
-                .setMessage("Project already exist with given short name")
-                .build();
-        throw StatusProto.toStatusRuntimeException(status);
+
+      List<String> accessibleProjectIds =
+          roleService.getAccessibleResourceIds(
+              null,
+              new CollaboratorUser(authService, userInfo),
+              ResourceVisibility.PRIVATE,
```

Comment:
Why do we need visibility in this listing?

Reply:
it is not used any where we can remove it. i will remove it.